app/assets/javascripts/reducers/course.js

```diff
@@ -52,7 +52,17 @@ export default function course(state = initialState, action) {
       const courseData = action.data.course;
       const newStats = CourseUtils.newCourseStats(state, courseData);
       const newKeys = CourseUtils.courseStatsToUpdate(courseData, newStats);
-      return { ...state, ...newKeys, newStats };
+      const refreshedLogs = CourseUtils.refreshCourseUpdatedAtTimestamp(state, courseData);
+
+      return {
+        ...state,
+        ...newKeys,
+        newStats,
+        flags: {
```

Comment:
This seems more complicated than it needs to be. Maybe it would be better to simply replace the entire `flags`?

I think the reason we limit the other changes to just the stats is to avoid overwriting unsaved changes that the user might be making to the course details. But I think the flags data can't be directly changed on the frontend, so it should be safe to replace the entire flags object with whatever is received from the server.

Reply:
> Maybe it would be better to simply replace the entire `flags`?

Okay, I will implement this. It sounds much simpler.

Reply:
Thanks! I assume you confirmed it works like this?

Reply:
> Thanks! I assume you confirmed it works like this?

Yes, it works properly. I have attached a screen recording below.


https://github.com/user-attachments/assets/4cd2c741-7e1b-44b1-94b2-388e5eb7b7a5